NetCDF preview

tcnichol opened the pull request:
Here is the netCDF preview. There is also a pull request in extractors-geo but wanted to put it here as well since it is a clowder2-demo. 

I checked in a sample file to test that was smaller, larger files are shared here:

https://drive.google.com/drive/folders/1ouc-jq8L21uwQzHNvP2VRrePxec5lO9i

longshuicy commented on preview.netcdf/extractor_info.json:
```diff
@@ -0,0 +1,46 @@
+  ],
+  "process": {
+    "file": [
+      "*/"
```
Does that mean it triggers on any file? Shouldn't it be `application/x-netcdf`?

tcnichol replied:
fixed

longshuicy commented on preview.netcdf/ncsa.geo.netcdf.extractor.py:
```diff
@@ -0,0 +1,82 @@
+plt.rcParams['figure.figsize'] = (16.0, 12.0)
+
+
+
```
Sorry for nitpicking... there are two many blank lines here. Could you try "black" or IDE reformatting

tcnichol replied:
did formatting

longshuicy commented on preview.netcdf/requirements.txt:
```diff
@@ -0,0 +1,7 @@
+gsconfig-py3==1.0.8
+pika>=1.0.0
```
is `pika` and `wheel` required?

longshuicy commented on preview.netcdf/requirements.txt:
```diff
@@ -0,0 +1,7 @@
+pika>=1.0.0
+requests>=2.10.0
+wheel>=0.24.0
+pyclowder
```
You might want to specify pyclowder to be >=3.0.7 Just to be safe?

tcnichol replied:
fixed clowder version

longshuicy reviewed (changes requested):
Im getting ModuleNotFoundError: No module named 'mpl_toolkits.basemap'
Does this library need to be included in requirement.txt?

longshuicy commented on preview.netcdf/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+RUN pip install -r requirements.txt
+
+COPY netcdfutils.py ncsa.geo.netcdf.extractor.py extractor_info.json ./
+CMD python ncsa.geo.netcdf.extractor.py
```
No need to pass in CLOWDER_VERSION as parameters here?

tcnichol replied:
this is added

tcnichol commented:
> Im getting ModuleNotFoundError: No module named 'mpl_toolkits.basemap' Does this library need to be included in requirement.txt?

I did not get that error locally, but I notice it showed up in the Docker image when I tried to run.

Since I was having dependency issues running things related to NetCDF in Docker I am switching to using a conda environment. Seeing if I can resolve this.

ddey2 reviewed (changes requested):
Can you add netcdfutils, matplotlib packages? I am seeing a lot of imports like netCDF4, mpl_toolkits etc which seems like not being used.

tcnichol commented:
As of now, I have been unable to get this to run in PyCharm using the Docker image as the run time environment. Looking into this.

In the meantime, it does work if you use the conda environment. I have also removed requirements.txt and replaced it with the environment.yml for the conda environment.

tcnichol commented:
> Can you add netcdfutils, matplotlib packages? I am seeing a lot of imports like netCDF4, mpl_toolkits etc which seems like not being used.

This should be changed in the environment.yml file now.
